packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -28,24 +29,27 @@ import 'platform_messages.dart';
 /// channels will interfere with each other's communication.
 ///
 /// See: <https://flutter.io/platform-channels/>
-class BasicMessageChannel<T> {
-  /// Creates a [BasicMessageChannel] with the specified [name] and [codec].
+class BaseMessageChannel<T> {
+  /// Creates a [BaseMessageChannel] with the specified [name], [codec] and [binaryMessenger].
   ///
-  /// Neither [name] nor [codec] may be null.
-  const BasicMessageChannel(this.name, this.codec);
+  /// None of [name], [codec], or [binaryMessenger] may be null.
+  const BaseMessageChannel(this.name, this.codec, this.binaryMessenger);
```

Comment:
Can we assert that the inputs are not null?

Reply:
Done.